src/js/store/listeners.js

```diff
@@ -91,31 +91,6 @@ function updateSelectedMarkInVega(selectedMark, vegaView) {
 }
 
 /**
- * Setting the same value on a signal twice should have no effect, so  we
- * naively set all signals on each store update to ensure store and Vega
- * stay in sync. Altering this to be smarter is one area to investigate should
- * any performance issues crop up later on, but signal writes are pretty fast.
- *
- * @param {Object} state - An Immutable state object
- * @param {Object} vegaView - A vega view instance to update with the selected mark
- * @returns {void}
- */
-function updateAllSignals(state, vegaView) {
```

Comment:
Since we now always update _both_ vega and the store with `sg.set`, we don't need this sync step anymore.